Doc updates

Juanlu001 opened the pull request:
Featuring new installation instructions (thanks to the work of @anhiga and myself) and two new success stories (courtesy of @nayeli173 and Yufeng Luo).

anhiga commented on docs/source/getting_started.rst:
```diff
@@ -21,25 +21,48 @@ Installation
-Development installations are also supported thanks to setuptools::
+  $ pip install https://github.com/poliastro/poliastro/archive/master.zip
 
-  $ pip install -e .
```
I would keep development mode installation instructions, either here or in `Contributing`.

Juanlu001 replied:
Good point. This should be part of a developer guide (see https://github.com/poliastro/poliastro/issues/176) but for the moment I agree to move it to `Contributing`.

Juanlu001 commented:
@anhiga Updated `CONTRIBUTING`, tell me what you think!

anhiga commented on CONTRIBUTING.rst:
```diff
@@ -82,3 +82,19 @@ coverage rate should increase or stay the same. Automatic services will ensure
+2. `Register to GitHub <https://github.com/>`_.
+3. `Fork poliastro <https://help.github.com/articles/fork-a-repo/>`_.
+4. `Clone your fork <https://help.github.com/articles/cloning-a-repository/>`_.
+5. Install it in development mode using :code:`pip install --editable .` (this
```
Maybe I would change `pip install --editable .` to `pip install --editable /path/to/cloned/repo` or something like that. Actually I'm being a little nitpicking (it's my first review, I have to write something), I really like all of the changes 😄

Juanlu001 commented:
What about now @anhiga? :grin:

anhiga commented:
All seems in order to me. 😄

dpshelio commented on CONTRIBUTING.rst:
```diff
@@ -82,3 +82,20 @@ coverage rate should increase or stay the same. Automatic services will ensure
+2. `Register to GitHub <https://github.com/>`_.
+3. `Fork poliastro <https://help.github.com/articles/fork-a-repo/>`_.
+4. `Clone your fork <https://help.github.com/articles/cloning-a-repository/>`_.
+5. Install it in development mode using
```
Would you like to have an environment (either conda's or python's)? That would make you able to test the "stable" versus the "master" and not mess up with other stuff you would have. We [explain conda's way](http://docs.sunpy.org/en/latest/dev_guide/version_control.html#installing-sunpy) in sunpy.

Juanlu001 replied:
Awesome! I might copy parts of that document :)

dpshelio commented on CONTRIBUTING.rst:
```diff
@@ -82,3 +82,20 @@ coverage rate should increase or stay the same. Automatic services will ensure
+   :code:`pip install --editable /path/to/poliastro` (this means that the
+   installed code will change as soon as you change it in the download
+   location).
+6. Make changes and commit.
```
I would create a branch first, otherwise people are going to get confused with `master` branches everywhere.

Juanlu001 replied:
You are right, I should note this.

dpshelio commented on docs/source/getting_started.rst:
```diff
@@ -21,25 +21,48 @@ Installation
+and OS X using pip::
+
+  $ pip install numpy  # Run this one first!
+  $ pip install scipy  # If this does not work, use conda
```
Don't you have a `requirements.txt` file? should not that install all for you?

Juanlu001 replied:
No we don't. As far as I know, they are not meant for libraries:

https://caremad.io/posts/2013/07/setup-vs-requirement/

Also, one has to install NumPy before everything because otherwise the world breaks, as `setup_requires` is not well supported.

https://github.com/pypa/setuptools/issues/377

dpshelio replied:
@Cadair - is that still true? we do have various different `requirement.txt` - what would you think should go here?

dpshelio replied:
Also, how does pypi knows the dependencies when installing libraries? is because the `setup.py`?

dpshelio replied:
👍 for the links!! I've learnt something new :) - Thanks

Juanlu001 replied:
Yes, PyPI gets the dependencies from `setup.py`! Using a requirements file is a common misconception :)

dpshelio commented on docs/source/success.rst:
```diff
@@ -1,6 +1,30 @@
 ---------------
 
+    *"My team and I used Poliastro for our final project in our Summer App Space
+    program. This module helped us in plotting asteroids by using the data
```
Maybe for the history of this you should say which year this happened.

Juanlu001 replied:
Yeah, the year is written in parentheses at the end!

dpshelio replied:
👍

Juanlu001 commented:
Note to self: add beyond (GPLv3), by @galactics, to similar projects :rocket:

Juanlu001 commented:
Add also SpiceyPy (MIT :tada:), by @AndrewAnnex, to similar projects :artificial_satellite:

Juanlu001 commented:
And also: a link to the Riot chat! :scream:

Juanlu001 commented:
More notes to self: regenerate plots... And if using an [alternative style](https://gist.github.com/Juanlu001/edb2bf7b583e7d56468a), document it in the developer/contributor guide.

Juanlu001 commented:
This is ready for the last review. I addressed all the comments, reviewed the whole user guide and even made one little but beautiful change to the code. When this gets merged half a dozen issues will close, so please give the green light :green_heart:

Juanlu001 commented:
@anhiga Would you please do a final review of this before merging?

anhiga commented on CONTRIBUTING.rst:
```diff
@@ -27,6 +27,21 @@ any inconsistency or opportunity for improvement, you can edit those too.
+
+    $ cd build/html
+    $ python -m http.server
+    Serving HTTP on 0.0.0.0 port 8000 ...
```
It doesn't render really well: 
![imagen](https://user-images.githubusercontent.com/18404378/29772633-f283c8ac-8bf9-11e7-82bc-05c8bdf9219f.png)

Maybe adding `$ Out: Serving HTTP on 0.0.0.0 port 8000 ...` would solve the problem, although it is not how linux shells show output.

Juanlu001 replied:
Only one colon missing, should have been `run::` :) Fixing!

anhiga commented on docs/source/user_guide.rst:
```diff
@@ -35,14 +41,20 @@ position and velocity vectors we can use
 And that's it! Notice a couple of things:
 
 * Defining vectorial physical quantities using Astropy units is very easy.
   The list is automatically converted to a :code:`Quantity`, which is actually
```
I would take advantage of `intersphinx`, and use
```
:py:mod:`astropy.units.Quantity`
```

anhiga commented on docs/source/user_guide.rst:
```diff
@@ -221,16 +268,18 @@ we can query the position and velocity of the planets of the Solar System.
 The function :py:func:`poliastro.ephem.get_body_ephem` will return
 position and velocity vectors using low precision ephemerides available in
-Astropy and an ``astropy.time.Time``::
+Astropy and an ``astropy.time.Time``:
```
Intersphinx, same as above.

Juanlu001 commented:
Addressed all the comments. Merging! :tada:
